Remotion/Data/DomainObjects.UnitTests/Infrastructure/ObjectIDStringSerialization/ObjectIDStringSerializerTest.cs

```diff
@@ -277,6 +277,44 @@ public void Parse_WithInvalidClassID_ThrowsFormatException ()
               "Serialized ObjectID 'Arder|5d09030c-25c2-4735-b514-46333bd28ac8|System.Guid' is invalid: 'Arder' is not a valid class ID."));
     }
 
+    [TestCase("Official|5d09030c-25c2-4735-b514-46333bd28ac8|System.Guid, mscorlib", typeof(Guid))]
+    [TestCase("Official|5d09030c-25c2-4735-b514-46333bd28ac8|System.Guid, mscorlib, Version=4.0.0.0", typeof(Guid))]
+    [TestCase("Official|23|System.Int32, mscorlib", typeof(Int32))]
+    [TestCase("Official|23|System.Int32 , mscorlib , Version=4.0.0.0", typeof(Int32))]
+    [TestCase("Official|test|System.String, mscorlib", typeof(String))]
+    [TestCase("Official|test| System.String , mscorlib , Version=4.0.0.0", typeof(String))]
+    [Test]
+    public void Parse_ExplicitAssemblyName (string idString, Type type)
+    {
+#if NET
```

Comment:
Please test for NETFRAMEWORK, not NET, since that's what we test for in the rest of the code, too.